fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/statemachine/ReplicaStateMachine.java

```diff
@@ -263,7 +265,9 @@ private void doHandleStateChanges(
 
                 // then, may remove the offline replica from isr
                 Map<TableBucketReplica, LeaderAndIsr> adjustedLeaderAndIsr =
-                        doRemoveReplicaFromIsr(validReplicas);
+                        checkIfDropTableOrPartitionRequest(validReplicas)
+                                ? doBatchRemoveReplicaFromIsr(validReplicas)
```

Comment:
Ask a question: Why don’t we design this to be more generic?  Instead of identifying specific operations (e.g., create table/partition, drop table/partition), why we don't merging buckets by table/partition and perform batch registration sequentially. If the cluster experiences significant ISR adjustment (e.g., frequent pod shutdown/starting), would batch registration become infeasible?